Applications/Terminal/main.cpp

```diff
@@ -207,6 +207,19 @@ static RefPtr<GUI::Window> create_settings_window(TerminalWidget& terminal)
         terminal.set_scroll_length(value);
     };
 
+    auto& history_size_spinbox_container = settings.add<GUI::GroupBox>("Maximum scrollback history lines");
+    history_size_spinbox_container.set_layout<GUI::VerticalBoxLayout>();
+    history_size_spinbox_container.layout()->set_margins({ 6, 16, 6, 6 });
+    history_size_spinbox_container.set_size_policy(GUI::SizePolicy::Fill, GUI::SizePolicy::Fixed);
+    history_size_spinbox_container.set_preferred_size(100, 46);
+
+    auto& history_size_spinbox = history_size_spinbox_container.add<GUI::SpinBox>();
+    history_size_spinbox.set_range(512, 40960);
```

Comment:
I guess a min value of 0 to disable scrollback would be nice? :)

Reply:
what sort of madman would do that...?

Reply:
are you honestly serious about this? I don't mind doing that, but it sounds like crazy talk to me

Reply:
don't feel like you have to implement it if the current solution doesn't allow it, but enforcing a limit on the lower end seems rather arbitrary :shrug:

Reply:
I'll let it do that if setting it to zero doesn't crash anything, I guess.
let's see...

Reply:
Ah, whatever, if the user wants to disable scrollback, they should be able to :shrug: